actions/ql/lib/codeql/actions/config/ConfigExtensions.qll

```diff
@@ -77,3 +77,8 @@ extensible predicate untrustedGitCommandDataModel(string cmd_regex, string flag)
  * Holds for gh commands that may introduce untrusted data
  */
 extensible predicate untrustedGhCommandDataModel(string cmd_regex, string flag);
+
+/**
+ * Holds if `action` needs `permission` to run.
```

Comment:
Expand this to link to the GitHub docs, and provide an example. I think we should explain that `action` will be the action name as used in a workflow (without the version) and `permission` will have the form `scope-name: read|write`.

Reply:
On versioning: I agree it would be useful to consider versions -- first-party actions often change scopes with major version bumps -- but that complicates how we model. We would need to list permissions for each action together with a minimum version that uses those permissions, and then parse and compare the versions seen in the workflow. I think that can be a follow-up design and change.